### ISSUE TITLE
[Telemetry] Sends different values of component type on odo create and push

### ISSUE THREAD

**Issue by valaparthvi** [openshift/odo#4928]
/kind bug

## What versions of software are you using?

**Operating System:**
Fedora 34
**Output of `odo version`:**
odo v2.2.3 (527c4c76b)
## How did you run odo exactly?
1. `odo create java-quarkus`
2. `odo push`

## Actual behavior
1. Check the component type value sent to Segment/Woopra, it shows `java-quarkus`.
2. Check the component type value sent to Segment/Woopra, it shows `quarkus`.
Both the values are different.

## Expected behavior
1. Check the component type value sent to Segment/Woopra, it shows `quarkus`.
2. Check the component type value sent to Segment/Woopra, it shows `quarkus`.
Both the values must be same.

**Comment by valaparthvi** [comment]
cc: @kadel

**Comment by valaparthvi** [comment]
Acceptance Criteria/ Scope of the issue:
1. - [ ] `odo push` and `odo create` should send the same value to Segment. See the description for an example.
2. - [ ] The component type must be a `.metadata.projectType`, if that is not available, then it can be `.metadata.Language`, and if neither of those are available, the value must be `Not available`.

----
The way it works for `odo create` is a little complicated than it does for `odo push`, but the only place where we need to make a change is `odo create`.

**odo push**
The `componentType` value is set during `odo push` from the devfile metadata, if a devfile exists, if not, it utilizes local config to identify this value.

But this is not where we need to make a change.

**odo create**
For S2I components, it's simpler; but for devfile components, we add a value here before the `devfileRun` method is called to ensure that we capture the value in case there is a failure at a later stage. At a later stage, when the devfile has been downloaded, we check if a value was assigned earlier, if not, we assign another value to the `componentType`.
```go
        // This is during the initial stage when `Run` method is called
	if !co.forceS2i && co.devfileMetadata.devfileSupport {
		if scontext.GetTelemetryStatus(cmd.Context()) {
			scontext.SetComponentType(cmd.Context(), co.devfileMetadata.componentType)
		}
                ...
        }
        // At a later stage, after the devfileData has been parsed. This is when `devfileRun` method is called
	if scontext.GetTelemetryStatus(cmd.Context()) {
		if value, ok := scontext.GetContextProperties(cmd.Context())[scontext.ComponentType]; !ok || value == "" {
			scontext.SetComponentType(cmd.Context(), component.GetComponentTypeFromDevfileMetadata(devObj.Data.GetMetadata()))
		}
	}
```
The way it works for devfile is where we need to change. We could either take the risk of not adding a value to `componentType` during the initial stage, which would mean that if the command execution fails before it can assign the value from metadata, we don't get to see which `componentType` the command failed for. Another way would be, we don't add the initial value and 
rely on it being added once the devfile is created.(WIP)

**Comment by dharmit** [comment]
@valaparthvi Is this good to be `triage/ready`?

**Comment by kadel** [comment]
Because we have different information available at different stages (`odo create` vs `odo push`) we can do things a little differently.
Main problem with the current implementation is that it uses the same field to store a different value.

**Let's keep the `componentType` fields as it is to not break the current data.**

On top of that lets add new fields:

For `odo create`
We want to capture what devfile user requested and we want to capture that even if they try to use non-existing devfile.

If user executed `odo create something` the telemetry event should have
- `devfileName: something`
If user executed `odo create` the telemetry event should have 
- `devfileName: Not Available`
In other words the `devfileName` should match the first argument

For `odo push`
- if  `metadata.language` in devfile is not empty send  its value in `language` field
- if `metadata.projectType` in devfile is not empty send  its value in `projectType` field
